File: src/com/codingspezis/android/metalonly/player/utils/BitmapDisplayer.java
package com.codingspezis.android.metalonly.player.utils;

import android.graphics.*;

import com.codingspezis.android.metalonly.player.utils.ImageLoader.PhotoToLoad;

//Used to display bitmap in the UI thread
class BitmapDisplayer implements Runnable {
	private final ImageLoader imageLoader;

	Bitmap bitmap;

	PhotoToLoad photoToLoad;

	public BitmapDisplayer(ImageLoader imageLoader, Bitmap b, PhotoToLoad p) {
		this.imageLoader = imageLoader;
		bitmap = b;
		photoToLoad = p;
	}

	@Override
	public void run() {
		if (this.imageLoader.imageViewReused(photoToLoad)) {
			return;
		}

		if (bitmap != null) {
			photoToLoad.imageView.setImageBitmap(bitmap);
		} else {
			photoToLoad.imageView.setImageResource(this.imageLoader.stub_id);
		}
	}
}

File: app/src/main/kotlin/com/codingspezis/android/metalonly/player/utils/BitmapDisplayer.kt
package com.codingspezis.android.metalonly.player.utils

import android.graphics.Bitmap
import com.codingspezis.android.metalonly.player.R

/**
 * Class to bind a Bitmap to an ImageView on the UIThread
 */
internal class BitmapDisplayer(private val imageLoader: ImageLoader, var bitmap: Bitmap?, var modImageLoadingQueueItem: ModImageLoadingQueueItem) : Runnable {

    override fun run() {
        if (this.imageLoader.imageViewReused(modImageLoadingQueueItem)) {
            return
        }

        if (bitmap != null) {
            modImageLoadingQueueItem.imageView.setImageBitmap(bitmap)
        } else {
            modImageLoadingQueueItem.imageView.setImageResource(R.drawable.mo_wait)
        }
    }
}